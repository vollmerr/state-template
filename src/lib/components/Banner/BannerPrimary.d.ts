import * as React from 'react';

export interface BannerPrimaryProps {
  /** Image source */
  src: string,
}

declare class BannerPrimary extends React.Component<BannerPrimaryProps, {}> {}

export default BannerPrimary;
