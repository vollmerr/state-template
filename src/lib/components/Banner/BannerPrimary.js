import React from 'react';
import T from 'prop-types';

const BannerPrimary = ({ src }) => (
  <div
    className={'header-primary-banner hidden-print'}
    style={{
      backgroundImage: `url(${src})`,
      height: '450px',
    }}
  />
);

BannerPrimary.propTypes = {
  src: T.string.isRequired,
};

export default BannerPrimary;
