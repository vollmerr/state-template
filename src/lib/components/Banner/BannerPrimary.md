```jsx
  const src = goldenGate; // point to image location here

  <BannerPrimary src={src} />
```
