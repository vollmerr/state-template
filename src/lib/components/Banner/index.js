export { default as BannerPrimary } from './BannerPrimary';
