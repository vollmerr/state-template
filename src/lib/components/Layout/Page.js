import React from 'react';
import T from 'prop-types';

import { BannerPrimary } from '../Banner';
import Container from './Container';
import Section from './Section';
import MainContent from './MainContent';
import MainPrimary from './MainPrimary';

const Page = ({
  bannerSrc,
  children,
  className,
  title,
}) => (
  <>
    {bannerSrc && <BannerPrimary src={bannerSrc} />}

    <MainContent className={className}>
      <MainPrimary>
        {
          title && (
          <Section variant={'standout'}>
            <Container>
              <h1 className={'m-b-0'}>{title}</h1>
            </Container>
          </Section>
          )
        }

        <Section>
          <Container>
            {children}
          </Container>
        </Section>
      </MainPrimary>
    </MainContent>
  </>
);

Page.propTypes = {
  /** Banner image src */
  bannerSrc: T.string,

  /** Page content */
  children: T.node,

  /** Class name to attach to main content */
  className: T.string,

  /** Page title  */
  title: T.string,
};

Page.defaultProps = {
  bannerSrc: null,
  children: null,
  className: null,
  title: null,
};

export default Page;
