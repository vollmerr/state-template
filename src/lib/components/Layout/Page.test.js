import React from 'react';
import { shallow } from 'enzyme';

import { Page as fromLib } from '../..';
import { BannerPrimary } from '../Banner';
import Page from './Page';

const props = {
  bannerSrc: 'test-banner-src',
  children: 'test-children',
  className: 'test-class',
  title: 'test-title',
};

let wrapper;
describe('Page', () => {
  beforeEach(() => {
    wrapper = shallow(<Page {...props} />);
  });

  it('should be exported as part of the library', () => {
    expect(fromLib).toEqual(Page);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should not render a banner if `bannerSrc` is not provided', () => {
    wrapper.setProps({ bannerSrc: null });
    expect(wrapper.find(BannerPrimary).length).toBe(0);
  });

  it('should not render a title if `title` is not provided', () => {
    wrapper.setProps({ title: null });
    expect(wrapper.find('h1').length).toBe(0);
  });
});
