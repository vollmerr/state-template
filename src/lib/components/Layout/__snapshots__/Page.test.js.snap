// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page should render correctly 1`] = `
<Fragment>
  <BannerPrimary
    src="test-banner-src"
  />
  <MainContent
    className="test-class"
  >
    <MainPrimary>
      <Section
        variant="standout"
      >
        <Container>
          <h1
            className="m-b-0"
          >
            test-title
          </h1>
        </Container>
      </Section>
      <Section
        variant={null}
      >
        <Container>
          test-children
        </Container>
      </Section>
    </MainPrimary>
  </MainContent>
</Fragment>
`;
